controller/api/destination/endpoint_translator.go

```diff
@@ -428,5 +428,5 @@ func getInboundPort(podSpec *corev1.PodSpec) (uint32, error) {
 			return uint32(port), nil
 		}
 	}
-	return 0, fmt.Errorf("failed to find %s environment variable in any container for given pod spec", envInboundListenAddr)
```

Comment:
Is this change necessary?

Reply:
Yep we don't want an error returned [here](https://github.com/linkerd/linkerd2/pull/6889/files#diff-6598a7b83bed9beb2b591cedbac5b5c052a813553d4e38437494128168d7a438R391). This means the destination pod is not injected, so we don't have a proxy port to set on the protocol hint [here](https://github.com/linkerd/linkerd2/pull/6889/files#diff-281e603e6940b45931c7339138278cec5c31c347e4d3a8d336b72d9c38d0f4a1R58).

Reply:
I considered making this more explicit by having `toWeightedAddr` return an additional `bool` that says if the pod is injected or not—something like `addr, injected, err := toWeightedAddr(...)`. Then when we create the `Endpoint`, we'd add a third field to it:
```
type Endpoint struct {
	addr      *pb.WeightedAddr
	proxyPort uint32
        injected bool
}
```

Finally, we'd use `endpoint.injected` instead of `endpoint.proxyPort != 0` as the check.

I decided against that though because I think it's enough to assume that if `proxyPort == 0`, the endpoint is not injected.

Reply:
where do we test for proxyPort == 0?

Reply:
I can update this to clarify the return, but `toWeightedAddr` hasn't changed as part of this.